ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```diff
@@ -0,0 +1,287 @@
+            for connected_endpoint in connected_endpoints:
+                if "adapters" not in connected_endpoint:
+                    continue
+
+                connected_endpoint["adapters"] = [adapter for adapter in connected_endpoint["adapters"] if self._hostname in adapter.get("switches", [])]
+
+                for adapter_index, adapter in enumerate(connected_endpoint["adapters"]):
+                    if "profile" not in adapter:
+                        continue
+
+                    port_profile = get_item(self._merged_port_profiles, "profile", adapter["profile"], default={})
+                    adapter = merge(port_profile, adapter, list_merge="replace", destructive_merge=False)
+                    adapter.pop("profile")
+                    connected_endpoint["adapters"][adapter_index] = adapter
+
+                connected_endpoint["type"] = connected_endpoints_key["type"]
+                filtered_connected_endpoints.append(connected_endpoint)
+
+        return filtered_connected_endpoints
+
+    @cached_property
+    def _filtered_network_ports(self) -> list:
+        """
+        Return list of endpoints defined under "network_ports"
+        which are connected to this switch.
+        """
+        filtered_network_ports = []
+        for network_port in get(self._hostvars, "network_ports", default=[]):
+            if not self._match_regexes(network_port.get("switches"), self._hostname):
+                continue
+
+            if "profile" in network_port:
+                port_profile = get_item(self._merged_port_profiles, "profile", network_port["profile"], default={})
+                network_port = merge(port_profile, network_port, list_merge="replace", destructive_merge=False)
+                network_port.pop("profile")
+
+            filtered_network_ports.append(network_port)
+
+        return filtered_network_ports
+
+    def _match_regexes(self, regexes: list, value: str) -> bool:
+        for regex in regexes:
+            if re.match(rf"^{regex}$", value):
+                return True
+
+        return False
+
+    @cached_property
+    def _port_profiles(self) -> list:
+        """
+        Return list of "port_profiles"
+        """
+        return convert_dicts(get(self._hostvars, "port_profiles", default=[]), "profile")
+
+    @cached_property
+    def _merged_port_profiles(self) -> list:
+        """
+        Return list of merged "port_profiles" where "parent_profile" has been applied.
+        """
+        merged_port_profiles = []
+
+        for port_profile in self._port_profiles:
+            if "parent_profile" in port_profile:
+                parent_profile = get_item(self._port_profiles, "profile", port_profile["parent_profile"], default={})
+                port_profile = merge(parent_profile, port_profile, list_merge="replace", destructive_merge=False)
+                port_profile.pop("parent_profile")
+            merged_port_profiles.append(port_profile)
```

Comment:
ok so I just realised what confused me - the fact that we use `destructive_merge=False` and reuse the `port_profile` name in this line:

```
port_profile = merge(parent_profile, port_profile, list_merge="replace", destructive_merge=False)
```

Can we maybe call the variable differently - like `merged_port_profile` ? As this looks like we are changing in place the port profile and in the situation you describe:
`parent-a -> profile-a -> adapter` and `profile-a -> profile-b -> adapter`

The outcome could be different if the order in `_port_profiles` because if we update the profiles in place with merge 
1. `[parent-a, profile-a, profile-b]`
or
2. `[profile-b, profile-a, parent-a]`

In 1. the parent-a would be merged with parent-a first and then profile-b would be updated with the value of the merged profile-a

In 2. the profile-b would be updated with the values of unmerged profile-a

so it becomes order dependent.

Right now everything in the clarity of reading the code relies on this destructive_merge False flag.